cypress/integration/networking.spec.ts

```diff
@@ -3,49 +3,165 @@ import * as path from "path";
 import * as ipcStub from "../support/ipc-stub";
 import * as nodeManager from "../support/nodeManager";
 
-context("networking", () => {
+context("p2p networking", () => {
+  context("network status indicator", () => {
+    it(
+      "reacts to network state changes",
+      { defaultCommandTimeout: 8000 },
+      () => {
+        nodeManager.withTwoOnboardedNodes((node1, node2) => {
+          nodeManager.asNode(node1);
+          commands.pick("connection-status-offline").should("exist");
+          nodeManager.connectTwoNodes(node1, node2);
+          commands.pick("connection-status-online").should("exist");
+          nodeManager.asNode(node2);
+          commands.pick("connection-status-online").should("exist");
+        });
+      }
+    );
+  });
+
   it("replicates a project from one node to another", () => {
     commands.withTempDir(tempDirPath => {
-      nodeManager.withTwoConnectedNodes((node1, node2) => {
-        nodeManager.asNode(node1);
-        const newProjectPath = path.join(tempDirPath, "node1/new-project");
+      nodeManager.withTwoOnboardedNodes(
+        (maintainerNode, contributorNode) => {
+          nodeManager.connectTwoNodes(maintainerNode, contributorNode);
+          nodeManager.asNode(maintainerNode);
+          const newProjectPath = path.join(
+            tempDirPath,
+            "maintainerNode/new-project"
+          );
 
-        cy.exec(`mkdir -p ${newProjectPath}`);
+          cy.exec(`mkdir -p ${newProjectPath}`);
 
-        ipcStub.getStubs().then(stubs => {
-          stubs.IPC_DIALOG_SHOWOPENDIALOG.returns(newProjectPath);
-        });
-        commands.pick("new-project-button").click();
-        commands.pick("name").type("new-fancy-project.xyz");
+          ipcStub.getStubs().then(stubs => {
+            stubs.IPC_DIALOG_SHOWOPENDIALOG.returns(newProjectPath);
+          });
+          commands.pick("new-project-button").click();
+          commands.pick("name").type("new-fancy-project.xyz");
 
-        commands.pick("new-project").click();
-        commands.pick("new-project", "choose-path-button").click();
-        // Make sure UI has time to update path value from stub,
-        // this prevents this spec from failing on CI.
-        cy.wait(500);
+          commands.pick("new-project").click();
+          commands.pick("new-project", "choose-path-button").click();
+          // Make sure UI has time to update path value from stub,
+          // this prevents this spec from failing on CI.
+          cy.wait(500);
 
-        commands.pick("create-project-button").click();
+          commands.pick("create-project-button").click();
 
-        commands
-          .pick("project-screen", "header")
-          .contains("new-fancy-project")
-          .should("exist");
+          commands
+            .pick("project-screen", "header")
+            .contains("new-fancy-project")
+            .should("exist");
```

Comment:
Can’t we just use `createProjectWithFixture()`? Project creation is tested thoroughly elsewhere and this would reduce the amount code and make the tests faster.

Reply:
I think technically we could, but we wouldn't have a way to get to the original working directory of the project, which I was planning on using later on in the test to test commit replication.

Reply:
Ok. Could you create an issue for this?

Reply:
It's captured in https://github.com/radicle-dev/radicle-upstream/issues/1529 in these sections:
  - Forking/Checking out changes
  - Replicating changes

Reply:
I'm planning on working on the second half of #1529 in a separate PR because we first need to make checkouts in `--test` mode work before we can test those.